code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm

```diff
@@ -0,0 +1,1195 @@
+		if(affinity_nearby[L] >= 0)
+			T = get_step_to(holder, L, 1)
+		// Else wander away
+		else
+			T = get_step_away(holder, L)
+	// Didn't find affinity people nearby, copypasta from normal wandering.
+	// We don't call ..() because it'll perform some of the same work again and want to avoid that
+	if(!T)
+		if(prob(5))
+			tepholder.lay_down()
+			return
+		var/moving_to = 0 // Apparently this is required or it always picks 4, according to the previous developer for simplemob AI.
+		moving_to = pick(cardinal)
+		holder.set_dir(moving_to)
+		T = get_step(holder,moving_to)
+	// Finally do move if we actually found somewhere we'd like to go
+	if(T)
+		holder.IMove(T)
+
+/datum/ai_holder/simple_mob/teppi/handle_idle_speaking()
+	if(holder.resting)
+		return
+	..()
+
+/datum/ai_holder/simple_mob/teppi/baby/handle_idle_speaking()
+	if(holder.resting)
+		return
+	..()
+
+/datum/ai_holder/simple_mob/teppi/on_hear_say(mob/living/speaker, message)
+	var/mob/living/simple_mob/vore/alienanimals/teppi/T = holder
+	if(holder.client)
+		return
+	if(!speaker.client)
+		return
+	if(!T.teppi_adult)
+		return
+	var/speaker_affinity = T.affinity[speaker.real_name]
+	if(findtext(message, "lets go") || findtext(message, "come teppi") || findtext(message, "come [holder.name]"))
+		if(speaker == leader)
+			return
+		if(!leader)
+			if(speaker_affinity >= 100)
+				set_follow(speaker, follow_for = 10 MINUTES)
+				holder.visible_message("<span class='notice'>\The [holder] starts following \the [speaker]</span>","<span class='notice'>\The [holder] starts following you.</span>")
+				return
+		else
+			var/mob/living/L = leader
+			if(!can_see_target(L))
+				lose_follow()
+				if(speaker_affinity >= 100)
+					set_follow(speaker, follow_for = 10 MINUTES)
+					holder.visible_message("<span class='notice'>\The [holder] starts following \the [speaker]</span>","<span class='notice'>\The [holder] starts following you.</span>")
+					return
+			else if(speaker_affinity > T.affinity[L.real_name])
+				holder.visible_message("<span class='notice'>\The [holder] starts following \the [speaker]</span>","<span class='notice'>\The [holder] starts following you.</span>")
+				set_follow(speaker, follow_for = 10 MINUTES)
+				return
+			if(speaker_affinity == T.affinity[L.real_name])
+				lose_follow()
+				holder.visible_message("<span class='notice'>\The [holder] gives off an anxious whine.</span>")
+	if(findtext(message, "stop teppi") || findtext(message, "stay here") || findtext(message, "stop [holder.name]"))
+		if(leader == speaker)
+			lose_follow()
+			holder.visible_message("<span class='notice'>\The [holder] stops following \the [speaker]</span>","<span class='notice'>\The [holder] stops following you.</span>")
+			return
+
+/mob/living/simple_mob/vore/alienanimals/teppi/mutant
+
+/mob/living/simple_mob/vore/alienanimals/teppi/mutant/New()
+	teppi_mutate = TRUE
+	. = ..()
+
+//Custom teppi colors! For funzies.
+
+/mob/living/simple_mob/vore/alienanimals/teppi/cass
```

Comment:
Comment: You don't actually need to do the blank type definitions. The New() on the type implies them into existence.

This can be especially annoying sometimes like on z-shadows when I forgot to remove some proc on them, so they continued existing just because there was a single proc defined on their type despite not being mentioned anywhere else in code.